### PR TITLE
Actually update event workflow via API

### PR DESCRIPTION
**Pull request by KatrinIhler**
The docs allow setting processing info when updating an event via POST
to api/events/, but the processing info is never actually used. Now this
will be used to update scheduled events if they aren't recorded yet.

### CODE REVIEW

File: modules/kernel/src/main/java/org/opencastproject/kernel/security/RemoteUserAndOrganizationFilter.java [hunk 1]
```diff
@@ -164,7 +164,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
           if (!originalUser.hasRole(GLOBAL_ADMIN_ROLE)) {
             // if the original user did not have system privileges, the target user must not gain those, either.
-            for (String systemRole : GLOBAL_SYSTEM_ROLES) {
+            for (String systemRole : GLOBAL_SYSTEM_ROLES) { // This IS WRONG
```
Comment (lkiesow) [inline modules/kernel/src/main/java/org/opencastproject/kernel/security/RemoteUserAndOrganizationFilter.java]: If this is wrong… why is this here?

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java [hunk 2]
```diff
@@ -455,6 +455,41 @@ public Response updateEventMetadata(@HeaderParam("Accept") String acceptHeader,
           indexService.updateEventAcl(eventId, eventHttpServletRequest.getAcl().get(), externalIndex);
         }
 
+        if (eventHttpServletRequest.getProcessing().isSome()) {
+          if (event.isScheduledEvent() && !event.hasRecordingStarted()) {
```
Comment (lkiesow) [inline modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java]: Avoid unnecessary long conditions:

```suggestion
          if (!event.isScheduledEvent() || event.hasRecordingStarted()) {
            return RestUtil.R.badRequest("Processing can't be updated for events that are already uploaded.");
          }
```

File: modules/kernel/src/main/java/org/opencastproject/kernel/security/RemoteUserAndOrganizationFilter.java (hunk 1 above)
Comment (KatrinIhler) [reply modules/kernel/src/main/java/org/opencastproject/kernel/security/RemoteUserAndOrganizationFilter.java]: Because... I forgot to remove this. Whoops.

**Comment by KatrinIhler** [comment]
@lkiesow Fixed!
